src/EmailLexer.php

```diff
@@ -140,18 +144,20 @@ class EmailLexer extends AbstractLexer
     /**
      * The last matched/seen token.
      *
-     * @var array
+     * @var array|Token
      *
      * @psalm-suppress NonInvariantDocblockPropertyType
-     * @psalm-var array{value:string, type:null|int, position:int}
-     * @psalm-suppress NonInvariantDocblockPropertyType
+     * @psalm-var array{value:string, type:null|int, position:int}|Token<int, string>
      */
     public $token;
 
     /**
      * The next token in the input.
      *
-     * @var array{position: int, type: int|null|string, value: int|string}|null
+     * @var array|Token
```

Comment:
changing the type of a public property in a non-internal class is still a BC break.

Reply:
It's an inherited BC break. This public property is declared in the Lexer library and apparently just overridden here to make the static analysis happy. As long as you pin Lexer to v1 in your project, the type won't change for you.